docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,274 @@
+export PYSPARK_PYTHON=environment/bin/python
+```
+### Spark-Submit Script Preparation
+* Set the `archives` argument to the path of the archive which was sent from the other node;
+```bash
+--archives environment.tar.gz#environment \
+```
+* Set the `properties-file` argument to override spark configuration by BigDL configuration file;
+```bash
+--properties-file ${BIGDL_HOME}/conf/spark-bigdl.conf \
+```
+* Load and distribute required dependency libraries in cluster through setting the `py-files` argument as BigDL Python zip file and local dependent Python file;
+```bash
+--py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,model.py \
+```
+* We need to register the BigDL jars files to the classpath of both driver and executors on yarn cluster.
+```bash
+--conf spark.driver.extraClassPath=${BIGDL_HOME}/jars/* \
+--conf spark.executor.extraClassPath=${BIGDL_HOME}/jars/*
+```
+
+Once all preparation works are completed, you could submit and execute the program with `spark-submit` script as below:
+```bash
+spark-submit \
+    --properties-file ${BIGDL_HOME}/conf/spark-bigdl.conf \
+    --py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,model.py \
+    --conf spark.driver.extraClassPath=${BIGDL_HOME}/jars/* \
+    --conf spark.executor.extraClassPath=${BIGDL_HOME}/jars/* \
+    --master yarn \
+    --deploy-mode client \
+    --executor-memory 10g \
+    --driver-memory 10g \
+    --executor-cores 8 \
+    --num-executors 2 \
+    --archives environment.tar.gz#environment \
+    train.py
+```
+
+## Yarn-Cluster
+When running programs on yarn cluster mode with `spark-submit` script, you could refer to the following steps:
+### Spark-Submit Script Preparation
+* Set the `archives` argument to the path of the Conda archive, which will be upload to remote clusters(i.e. HDFS) and distributed between all executors.
+```bash
+--archives environment.tar.gz#environment
+```
+* For yarn-cluster mode, both driver and executors will use the Python interpreter in `environment.tar.gz`, 
+```bash
+--conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=environment/bin/python \
+--conf spark.executorEnv.PYSPARK_PYTHON=environment/bin/python
+```
+* Set the `py-files` argument to load and distributed BigDL and other dependency libraries on yarn cluster.
+```bash
+--py-files ${BIGDL_HOME}/python/bigdl-spark_${SPARK_VERSION}-${BIGDL_VERSION}-python-api.zip,model.py
+```
+* This tutorial is mainly based on a BigDL Orca example, please download jars files from [BigDL Dllib Jars](https://repo1.maven.org/maven2/com/intel/analytics/bigdl/bigdl-dllib-spark_2.4.6/2.0.0/bigdl-dllib-spark_2.4.6-2.0.0-jar-with-dependencies.jar) and [BigDL Orca Jars](https://repo1.maven.org/maven2/com/intel/analytics/bigdl/bigdl-orca-spark_2.4.6/2.0.0/bigdl-orca-spark_2.4.6-2.0.0-jar-with-dependencies.jar) separately, then set the `jars` argument as below to register and transfer the BigDL jars files to the cluster.
```

Comment:
can we simply use the BigDL assembly package for the JARs?

Reply:
Dllib and Orca jars in BigDL package is not working, they may lack of some dependences.

Reply:
we need to fix that @glorysdj @Le-Zheng

Reply:
As BigDL-core jars were split from dllib, it will lack dependencies if you only use Dllib and Orca jars.
Actually, we only need to download [BigDL assembly package](https://repo1.maven.org/maven2/com/intel/analytics/bigdl/bigdl-assembly-spark_3.1.2/2.0.0/bigdl-assembly-spark_3.1.2-2.0.0.zip) which contains all the modules of jars. Please unzip bigdl-assembly-spark_3.1.2-2.0.0.zip and set jars  `--jars bigdl-assembly/jars/*`

Reply:
> As BigDL-core jars were split from dllib, it will lack dependencies if you only use Dllib and Orca jars. Actually, we only need to download [BigDL assembly package](https://repo1.maven.org/maven2/com/intel/analytics/bigdl/bigdl-assembly-spark_3.1.2/2.0.0/bigdl-assembly-spark_3.1.2-2.0.0.zip) which contains all the modules of jars. Please unzip bigdl-assembly-spark_3.1.2-2.0.0.zip and set jars `--jars bigdl-assembly/jars/*`

We have tried this before, it will return "No main class set in JAR". And maybe loading all jars files could be too many for users if they only need orca & dllib jars and other few dependencies.